file-generating/ronn/ronn.def

```diff
@@ -0,0 +1,326 @@
+Bootstrap: library
+From: alpine:3.8
+
+%help
+for container help:
+  $ ./ronn.sif --run-help
+
+for container info:
+  $ ./ronn.sif --run-info
+
+%runscript
+if [ -z "$1" ] || [ "$1" = "--help" ] || [ "$1" = "--run-help" ]; then
+    cat /container/runscript.help
+    return 0
+fi
+if [ "$1" = "--info" ] || [ "$1" = "--run-info" ]; then
+    cat /container/runscript.info
+    return 0
+fi
+if [ "$1" = "view" ] || [ "$1" = "view-file" ]; then
+    if [ ! -e "$2" ]; then
+        echo "$2: No such file or directory"
+        return 1
+    fi
+    # check if file is a directory
+    if [ -d "$2" ]; then
+        echo "$2: Is a directory"
+        return 2
+    fi
+    if [ "${2: -3}" = ".md" ] || [ "${2: -5}" = ".ronn"]; then
+        echo "ronn.sif: INFO: makeing tmp file..."
+        TMP_FILE="$2.tmp.1.ronn"
+        cat "$2" > "$TMP_FILE"
+        ronn -r "$TMP_FILE" >/dev/null 2>&1
+        rm -f "$TMP_FILE"
+        FILE=` echo "$2" | cut -f 1 -d '.' `
+        mandoc "./$FILE.1"
+        rm -f "$FILE.1"
+        return 0
+    fi
+    mandoc "./$2"
+    return 0
+elif [ "$1" = "ronn" ]; then
+    comd=` echo "$@" | cut -d ' ' -f2- `
+    ronn $comd
+    return 0
+elif [ "$1" = "system" ] || [ "$1" = "exec" ]; then
+    SYSTEM_COMMAND=` echo "$@" | cut -d ' ' -f2- `
+    $SYSTEM_COMMAND
+    return "$?"
+fi
+
+IS_OUT="false"
+
+check_args() {
+    if [ "$OPTION" = "-m" ] || [ "$OPTION" = "--man" ]; then
```

Comment:
Where are you getting OPTION from? Below you're doing:
```
for OPTION in "$@"; do
  check_args
done
```
But you aren't _passing_ OPTION ... and it _should_ be out of scope. 

IMO you should do:
```
check_args $OPTION
```

And in check_args do:
```
check_args() {
    OPTION = $1
    [ ... ]
```

Reply:
Yeah! thats what I meant to do.
Or we could put the whole `check_args()` function in the `for` loop, that may be a better idea... 

Again, I didin't spend very much time on this example, This was a "learn something cool instead of watching a movie", this was for my entertainment not really for usefulness.